pkg/analyze/ceph.go

```diff
@@ -102,16 +103,23 @@ type PgMap struct {
 func cephStatus(analyzer *troubleshootv1beta2.CephStatusAnalyze, getCollectedFileContents func(string) ([]byte, error)) (*AnalyzeResult, error) {
 	fileName := path.Join(collect.GetCephCollectorFilepath(analyzer.CollectorName, analyzer.Namespace), "status.json")
 	collected, err := getCollectedFileContents(fileName)
+	analyzeResult := &AnalyzeResult{}
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to read collected ceph status")
+
```

Comment:
I think we should differentiate other `error`s from `file-not-found` errors. `file-not-found` errors will be error logged in some cases (e.g what this PR proposes) while others will be propagated to the caller.

What you want to do is
- Create a new error type e.g `NotFoundError`
```go
type NotFoundError struct {
    Name string
}

func (e *NotFoundError) Error() string { return e.Name + ": not found" }

// Comparison
if errors.Is(err, NotFoundError) {
    // err, or some error that it wraps, is a not-found error
}
```
For more on custom errors read https://go.dev/blog/go1.13-errors 

- Return or wrap not-found errors as `NotFoundError` [here](https://github.com/replicatedhq/troubleshoot/blob/main/pkg/preflight/analyze.go#L55) and [here (you want to check file existence I believe, but maybe io.ReadFile returns an error that can be inspected)](https://github.com/replicatedhq/troubleshoot/blob/main/pkg/analyze/download.go#L298)
- Log not found errors and return `nil, nil`, but return all other errors like before. Remove `v.GetBool("debug")`, its setup by library clients

Reply:
Thank you @banjoh ! It is a great suggestion. I have added NotFoundError as type of new error and use it for file missing and other add-on not installed.

In #659, it required to hide on stdout for 
```
Longhorn: list nodes.longhorn.io: the server could not find the requested resource
rook ceph tools pod not found
```
So I added a block in collect to silent the error if it is `NotFoundError`.

However, I am not 100% sure that those changes did match you suggestions? Could you help me to review it again? If it matches, I will write those tests for it.